lib/ansible/modules/windows/win_domain_membership.ps1

```diff
@@ -100,17 +108,13 @@ Function Join-Domain {
     Param(
         [string] $dns_domain_name,
         [string] $new_hostname,
-        [string] $domain_admin_user,
-        [string] $domain_admin_password,
+        [pscredential] $domainCredential,
```

Comment:
Common PS syntax is to just use `credential`, I don't see a reason to specify a specific type of credential now that trying to move across to a more ps-like syntax.